### DISCUSSION TITLE
"MessageId" header missing in Spring Cloud AWS v3

### DISCUSSION THREAD

**Discussion by tjzeeman** [awspring/spring-cloud-aws#1219] [General]
While migrating our project from Spring Cloud AWS v2 to v3, I found that the new version no longer sets the "MessageId" header when converting from an AWS SDK Message to a Spring Message type. It caused issues with some of our error handlers that use `@Header("MessageId")` to retrieve the SQS message id. They now throw exceptions that this header is not present.

Looking at the new code, I see that "ReceiptHandle" is still being added as a header, albeit with a prefix. This was done in the same QueueMessageUtils class in v2 as the "MessageId" header.

I'm wondering if this was done intentionally and if I should switch to using the "id" header instead.
Or if it was perhaps missed in the new implementation?

**Comment by tjzeeman** [comment]
I updated to v3.2 this morning, did another search for this issue, and found that the two headers set in the QueueMessageUtils have moved into SqsHeaderMapper. There are breaking changes though.
The ReceiptHandle now has the prefix "Sqs_" so old messages won't be recognised. Not a problem in my case though.

More of an issue is the way the message-id from the AWS Message is passed on. It is still put into a header but via a route within the Spring Message structure and is now stored under the key "id" i.o. "MessageId". So a breaking change as well. 
This does allow for easy retrieval via message.getHeaders().getId(), and in my case, replacing the `@Header("MessageId")` with `@Header("id")`.

**Comment by tomazfernandes** [comment]
Hi @tjzeeman, you're correct, this is a complete rewrite of the solution and there are quite a few breaking changes.

Please let us know if you need any further assistance, thanks.
